lib/enforce_schema_rules.rb

```diff
@@ -71,10 +74,10 @@ def build_validation_args(options, col_type, validation_option = :invalid)
                       when :numeric
                         lambda { |col| col.name !~ exclusion_regexp && col.number? && col.type != :integer }
                       when :not_null
-                        # I have to exclude boolean types because of a "feature" of the way validates_presence_of 
+                        # I have to exclude boolean types because of a "feature" of the way validates_presence_of
                         # handles boolean fields
                         # See http://dev.rubyonrails.org/ticket/5090 and http://dev.rubyonrails.org/ticket/3334
-                        lambda { |col| (col.name !~ exclusion_regexp || col.name =~ /_id$/) && !col.null && col.type != :boolean }
```

Comment:
This is a good change. I excluded `_id` fields originally because if you aren't setting one it's typically a bug, and not something a user can fix on their end, but that's a pretty broad assumption that isn't always true.